File: latihanB/python/Pc.py
from sklearn.preprocessing import RobustScaler
from Processor import *
from Disk import *
from Ram import *

class Pc:
    def __init__(self, pObj, dObj, rObj):
        self.__processorObj = pObj
        self.__diskObj = dObj
        self.__ramObj = rObj
        self.__totalPrice = 0

    def setProcessor(self, pObj):
        self.__processorObj = pObj

    def getProcessor(self):
        return self.__processorObj

    def setDisk(self, dObj):
        self.__diskObj = dObj

    def getDisk(self):
        return self.__diskObj

    def setRam(self, rObj):
        self.__ramObj = rObj

    def getRam(self):
        return self.__ramObj

    def setTotalPrice(self, pPrice, dPrice, rPrice):
        self.__totalPrice = pPrice + dPrice + rPrice

    def getTotalPrice(self):
        return self.__totalPrice

    def printPc(self):
        print("-> Processor")
        self.__processorObj.printProcessor()
        print("-> Disk")
        self.__diskObj.printDisk()
        print("-> Ram")
        self.__ramObj.printRam()
        print("-> Total Price")
        self.setTotalPrice(self.__processorObj.getPrice(),
                            self.__diskObj.getPrice(),
                            self.__ramObj.getPrice())
        print("   " + str(self.__totalPrice) + " IDR")

File: latihanA/Pc.cpp
#include <bits/stdc++.h>
#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

using namespace std;

class Pc {
    private:
        Processor processorObj;
        Disk diskObj;
        Ram ramObj;
        int totalPrice;

    public:
        Pc() { }

        Pc(Processor pObj, Disk dObj, Ram rObj) {
            this->processorObj = pObj;
            this->diskObj = dObj;
            this->ramObj = rObj;
            this->totalPrice = 0;
        }

        void setProcessor(Processor pObj) {
            this->processorObj = pObj;
        }

        Processor getProcessor() {
            return this->processorObj;
        }

        void setDisk(Disk dObj) {
            this->diskObj = dObj;
        }

        Disk getDisk() {
            return this->diskObj;
        }

        void setRam(Ram rObj) {
            this->ramObj = rObj;
        }
        Ram getRam() {
            return this->ramObj;
        }

        void setTotalPrice(int pPrice, int dPrice, int rPrice) {
            this->totalPrice = pPrice + dPrice + rPrice;
        }

        int getTotalPrice() {
            return this->totalPrice;
        }

        void printPc() {
            cout << "-> Processor" << endl;
            this->processorObj.printProcessor();
            cout << "-> Disk" << endl;
            this->diskObj.printDisk();
            cout << "-> Ram" << endl;
            this->ramObj.printRam();
            this->setTotalPrice(this->processorObj.getPrice(), this->diskObj.getPrice(), this->ramObj.getPrice());
            cout << "-> Total Price" << endl;
            cout << "   " << this->totalPrice << " IDR" << endl;
        }
        
        ~Pc() { }
};